File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
// Prototipe for self calling function
int calc(int value, int base[], int i);
// This programm is a "greedy Algorithm" for calculate min number of coins needed for exchange
int main(void)
{
    int base[] = {25, 10, 5, 1};
    int i = 0;
    float money = 0;
    do
    {
        money = get_float("Enter how many money you have to exchange. Only 0 and positive numbers accepted: ");
    }
    while (money < 0);
    int value = round(money * 100);
    int change = 0;
    int remaider = 0;
    change = calc(value, base, i);
    printf("%d\n", change);
}
// Calculate integer number of coins using the array value as a basis
int calc(int value, int base[], int i)
{
    int change = 0;
    int remaider = 0;
    int x = value / base[i];
    if (x > 0)
    {
        change += x;
    }
    remaider = value % base[i];
    if (remaider > 0)
    {
        i++;
        change += calc(remaider, base, i);
    }
    return change;
}

File: pset6/cash/cash.py
from cs50 import get_float


def calc(value, base, i):
    change = 0
    remainder = 0
    x = value // base[i]
    if x > 0:
        change += x
    remainder = value % base[i]
    if remainder > 0:
        i += 1
        change += calc(remainder, base, i)
    return change


while True:
    money = get_float("Change owned: ")
    if money > 0:
        break
base = [25, 10, 5, 1]
i = 0
value = money * 100
change = calc(value, base, i)
# Explicit type convertion float to int
change = int(change)
print(change)